File: reservation-service/domain/availability.go
package domain

import (
	"time"

	"github.com/gocql/gocql"
)

type Availability struct {
	AvailabilityID gocql.UUID `json:"availabilityId"`
	AccommID       string     `json:"accommId"`
	Name           string     `json:"name"`
	Location       string     `json:"location"`
	MinCapacity    int        `json:"minCapacity"`
	MaxCapacity    int        `json:"maxCapacity"`
	StartDate      time.Time  `json:"startDate"`
	EndDate        time.Time  `json:"endDate"`
}

File: front-end-service/src/app/model/availability.ts
import { formatDate } from "@angular/common";

export class Availability{
    availabilityId: string;
    accommId: string;
    name: string;
    location: string;
    minCapacity: number;
    maxCapacity: number;
    startDate: Date;
    endDate: Date;
    constructor() {
        this.availabilityId = ""
        this.accommId = ""
        this.name = ""
        this.location = ""
        this.minCapacity = 0
        this.maxCapacity = 0
        this.startDate = new Date(formatDate(Date.now(),'yyyy-MM-dd','en_us'))
        this.endDate = new Date(formatDate(Date.now(),'yyyy-MM-dd','en_us'))
    }
}